content/blog/2022/2022-08-25-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,277 @@
+
+To begin, open your terminal and create a new folder.
+```
+mkdir node-cron-demo
+```
+
+Next change into the new project's directory
+```
+cd node-cron-demo
+```
+You will need to create a file index.js here, this is where we'll be writing all our codes.
+```
+touch index.js
+```
+Now run the command below to initialize the project. This will generate a `package.json` file which you can use to keep track of all project dependencies.
+```
+npm init -y
+```
+
+Next, let's add node-cron and other modules used later in this post.
+```
+npm install node-cron node-mailer 
+```
+
+## What we will be building
+To demonstrate the functionality of the node-cron library, we will build `4` sample applications.
+
+### 1. Scheduling a simple task with node cron.
+Any task of your choosing can be automated and run at a specific time using cron job schedulers. In this section, we'll write a simple function that logs to the terminal at the our specified time.
+
+```javascript
+const cron = require("node-cron");
+const express = require("express");
+
+const app = express();
+
+cron.schedule("*/15 * * * * *", function () {
+  console.log("---------------------");
+  console.log("running a task every 15 seconds");
+  //replace simple tasks
+});
+
+app.listen(3000, () => {
+  console.log("Application Listening.....");
+});
+```
+Run `node index.js` in the terminal.
+
+`output`
+```
+Application Listening.....
+---------------------
+running a task every 15 second
+```
+
+
+### 2. Scheduling email using node cron
+Emailing is a common feature of modern applications. Cron jobs can be used to accomplish this; for instance, a job schedule can be set to automatically send users an email each month with the most recent information from a blog or a product.
+
+```javascript
+const express = require("express");
+const cron = require("node-cron");
+const nodemailer = require("nodemailer");
+app = express();
+
+//Send email after 1 minute
+cron.schedule("1 * * * *", function () {
+  mailService();
+});
+
+function mailService() {
+  let mailTransporter = nodemailer.createTransport({
+    service: "outlook",
+    auth: {
+      user: "<your-email>@gmail.com",
+      pass: "***********",
+    },
+  });
+
+  // Setting credentials
+  let mailDetails = {
+    from: "<your-email>@gmail.com",
+    to: "<user-email>@gmail.com",
+    subject: "Test Mail using Cron Job",
+    text: "Node.js Cron Job Email Demo Test from Reflectoring Blog",
+  };
+
+  // Sending Email
+  mailTransporter.sendMail(mailDetails, function (err, data) {
+    if (err) {
+      console.log("Error Occurs", err.message);
+    } else {
+      console.log("---------------------");
+      console.log("Email sent successfully");
+    }
+  });
+}
+
+app.listen(3000, () => {
+  console.log("Application Listening.....");
+});
+
+```
+>**N:B** Gmail has stopped third-party access read [here](https://support.google.com/mail/answer/185833?hl=en-GB) on how to sign in using app passwords 
```

Comment:
This text here is a bit out of context. What is "third-party access" and what does it have to do with me sending emails? Also, why is this formatted as a quote?

Does the code above actually work with Gmail? That would be awesome! Remove this sentence and at the end of the section celebrate it by mentioning that the reader can try it out and it should work on their machine if they have a Gmail account. Explain shortly what they have to do to make it work instead of just giving them a link to a seemingly random Google support page (I guess they would have to create an "app password")? Do include the link, but also explain it in your words.